### PR TITLE
Fix scaling for MQ-(C|R)NN when distribution outputs are used

### PR DESCRIPTION
**Pull request by dcmaddix**
*Issue #, if available:*
https://github.com/awslabs/gluon-ts/issues/1069

*Description of changes:*
- Updated default `scaling` for large to `True` to be used with `DistrOutput`
- Left default for `scaling_decoder_dynamic_feature` to be `False`
- Pass `scaling_decoder_dynamic_feature` to training and prediction networks
- Fix the shape of `loc` and `scale`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by lostella** [comment]
@dcmaddix thanks! I think scaling *must* be off when `quantile_output` is used, because apparently the output doesn’t get scaled back: so you don’t want to scale the input, otherwise for fixed model parameters and two series `ts` and `alpha*ts`, you will get the same predictions for any `alpha > 0`. To allow for scaling in this case, the output quantiles should be scaled back.

On the other hand, scaling appears to be necessary when using distributions, otherwise NaNs are all over the place unless data is nicely scaled (like all time series with the same magnitude or something).

So I ended up making the default value for `scaling` dependent on what is being used, which maintains backwards compatibility in the `quantile_output` case, and fixese the `distr_output` one. Plus I've fixed the shape issue described in #1069.

File: src/gluonts/model/seq2seq/_forking_network.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def __init__(
         embedding_dimension: List[int],
         distr_output: Optional[DistributionOutput] = None,
         quantile_output: Optional[QuantileOutput] = None,
-        scaling: bool = False,
+        scaling: bool = True,
```
Comment (dcmaddix) [inline src/gluonts/model/seq2seq/_forking_network.py:80]: Should we update the default for `scaling` back to `False` here?

**Comment by dcmaddix** [comment]
> @dcmaddix thanks! I think scaling _must_ be off when `quantile_output` is used, because apparently the output doesn’t get scaled back: so you don’t want to scale the input, otherwise for fixed model parameters and two series `ts` and `alpha*ts`, you will get the same predictions for any `alpha > 0`. To allow for scaling in this case, the output quantiles should be scaled back.
> 
> On the other hand, scaling appears to be necessary when using distributions, otherwise NaNs are all over the place unless data is nicely scaled (like all time series with the same magnitude or something).
> 
> So I ended up making the default value for `scaling` dependent on what is being used, which maintains backwards compatibility in the `quantile_output` case, and fixese the `distr_output` one. Plus I've fixed the shape issue described in #1069.
> 
> Edit: I had to bump the test accuracy for the `distr_output` cases, since now the default scaling behaviour changed.

**Comment by dcmaddix** [comment]
> @dcmaddix thanks! I think scaling _must_ be off when `quantile_output` is used, because apparently the output doesn’t get scaled back: so you don’t want to scale the input, otherwise for fixed model parameters and two series `ts` and `alpha*ts`, you will get the same predictions for any `alpha > 0`. To allow for scaling in this case, the output quantiles should be scaled back.
> 
> On the other hand, scaling appears to be necessary when using distributions, otherwise NaNs are all over the place unless data is nicely scaled (like all time series with the same magnitude or something).
> 
> So I ended up making the default value for `scaling` dependent on what is being used, which maintains backwards compatibility in the `quantile_output` case, and fixese the `distr_output` one. Plus I've fixed the shape issue described in #1069.
> 
> Edit: I had to bump the test accuracy for the `distr_output` cases, since now the default scaling behaviour changed.

Looks great, thanks for updating the default to cover both cases! :)

File: src/gluonts/model/seq2seq/_forking_estimator.py [hunk 2]
```diff
@@ -154,7 +155,7 @@ def __init__(
         enable_encoder_dynamic_feature: bool = True,
         enable_decoder_dynamic_feature: bool = True,
         trainer: Trainer = Trainer(),
-        scaling: bool = False,
+        scaling: Optional[bool] = None,
         scaling_decoder_dynamic_feature: bool = False,
```
Comment (dcmaddix) [inline src/gluonts/model/seq2seq/_forking_estimator.py:159]: Do we want to turn on scaling of the dynamic features or this should be unrelated to the distribution change right?

File: src/gluonts/model/seq2seq/_forking_network.py (hunk 1 above)
Comment (lostella) [reply src/gluonts/model/seq2seq/_forking_network.py:80]: Right!

File: src/gluonts/model/seq2seq/_forking_estimator.py (hunk 2 above)
Comment (lostella) [reply src/gluonts/model/seq2seq/_forking_estimator.py:159]: I think this is unrelated maybe, we could do it but maybe as a separate story

**Review by lostella — APPROVED** [review]
